index.html

```diff
@@ -1028,6 +1061,454 @@ <h2>
         </dd>
       </dl>
     </section>
+    <section data-dfn-for="GamepadHapticActuator" data-link-for="GamepadHapticActuator">
+      <h2>
+        <dfn>GamepadHapticActuator</dfn> Interface
+      </h2>
+      <p>
+        A {{GamepadHapticActuator}} corresponds to a configuration of motors or
+        other actuators that can apply a force for the purposes of haptic
+        feedback.
+      </p>
+      <pre class="idl">
+        [Exposed=Window]
+        interface GamepadHapticActuator {
+          readonly attribute FrozenArray&lt;GamepadHapticEffectType&gt; effects;
+          Promise&lt;GamepadHapticsResult&gt; playEffect(
+              GamepadHapticEffectType type,
+              optional GamepadEffectParameters params = {}
+          );
+          Promise&lt;GamepadHapticsResult&gt; reset();
+        };
+      </pre>
+      <p>
+        Instances of {{GamepadHapticActuator}} are created with the internal
+        slots described in the following table:
+      </p>
+      <table class="simple">
+        <tr>
+          <th>
+            Internal slot
+          </th>
+          <th>
+            Initial value
+          </th>
+          <th>
+            Description
+          </th>
+        </tr>
+        <tr>
+          <td>
+            <dfn data-dfn-for="GamepadHapticActuator">[[\effects]]</dfn>
+          </td>
+          <td>
+            An empty [=list=] of {{Gamepad/GamepadHapticEffectType}}.
+          </td>
+          <td>
+            Represents the effects supported by the actuator.
+          </td>
+        </tr>
+        <tr>
+          <td>
+            <dfn data-dfn-for=
+            "GamepadHapticActuator">[[\playingEffectPromise]]</dfn>
+          </td>
+          <td>
+            `null`
+          </td>
+          <td>
+            The {{Promise}} for the playing effect, or `null` if no effect is
+            playing.
+          </td>
+        </tr>
+      </table>
+      <dl data-dfn-for="GamepadHapticActuator">
+        <dt>
+          <dfn>effects</dfn> attribute
+        </dt>
+        <dd>
+          <p>
+            Array of {{Gamepad/GamepadHapticEffectType}} values representing all
+            the types of haptic effects that the actuator supports. This
+            property MUST contain all the {{Gamepad/GamepadHapticEffectType}}
+            values that the actuator supports, unless the [=user agent=] does
+            not support playing effects of that type. The same object MUST be
+            returned and MUST NOT change for the lifetime of the
+            {{GamepadHapticActuator}}.
```

Comment:
Moved the following to the IDL. 

```suggestion
            not support playing effects of that type.
```